### PR TITLE
Boyer Moore Algorithm in Js

### PR DESCRIPTION
**Pull request by PuravUdayDesai**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes.)

### Checklist

- [Y] I've read the contribution guidelines.
- [Y] I've checked the issue list before deciding what to submit.
- [Y] I've edited the `README.md` and link to my code.

### CODE REVIEW

**Comment by PuravUdayDesai** [comment]
@iamrajiv please review the pull request once, if any changes/improvements/modifications needed please let me know about the respective.
Issue Number: #1439 
Pull Request Number: #1454

File: JavaScript/README.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
 ## Algorithms
 - [Russian Peasant Algorithm](Algorithms/RussianPeasant.js)
 - [Z Algorithm](Algorithms/ZAlgorithm.js)
+- [Boyer Moore's Majority Voting Algorithm](Algorithms/BoyerMooreAlgorithm.js)
```
Comment (atarax665) [inline JavaScript/README.md]:
```suggestion
- [Boyer Moore's Majority Vote Algorithm](Algorithms/BoyerMooreAlgorithm.js)
```

File: JavaScript/Algorithms/BoyerMooreAlgorithm.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
+$(document).ready(() => {
```
Comment (atarax665) [inline JavaScript/Algorithms/BoyerMooreAlgorithm.js]: I don't think we need this jQuery method in this program, moreover the program won't be executed in many online Node.js IDE due to this.

File: JavaScript/Algorithms/BoyerMooreAlgorithm.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        } else {
+            console.log("No Majority Candidate Detected")
+        }
+    }
+
+    //Declaration of list of candidates
+    var listOfCandidates = new Array(3, 3, 3, 3, 1, 1, 1, 3, 2);
+    //Calls main function for starting execution process
+    main(listOfCandidates);
+
```
Comment (atarax665) [inline JavaScript/Algorithms/BoyerMooreAlgorithm.js:97]:
```suggestion
//Output : 3 Has The Majority Number Of Votes
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by PuravUdayDesai** [comment]
Ok I'll update the code according to the changes suggested

File: JavaScript/Algorithms/BoyerMooreAlgorithm.js (hunk 2 above)
Comment (PuravUdayDesai) [reply JavaScript/Algorithms/BoyerMooreAlgorithm.js]: Okay I'll update the respective.

**Comment by PuravUdayDesai** [comment]
@atarax665 I have update the code according to the changes suggested, please re-review it once and let me know if any other modifications required, thank you for your time and interest.

**Review by atarax665 — APPROVED** [review]
LGTM :+1:

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻

**Review by iamrajiv — APPROVED** [review]

**Comment by iamrajiv** [comment]
@all-contributors please add @PuravUdayDesai for code

**Comment by PuravUdayDesai** [comment]
Thank you so much

**Pull request merged**